File: Easy/Java/1700. Number of Students Unable to Eat Lunch.java
class Solution {
    public int countStudents(int[] students, int[] sandwiches) {
        int zeros = 0;
        int ones = 0;

        for(int n : students){
            if(n == 0)
                zeros++;
            else
                ones++;
        }

        for(int n : sandwiches){
            if((n == 0 && zeros == 0) || (n == 1 && ones == 0))
                break;
            if(n == 0)
                zeros--;
            else
                ones--;
        }

        return Math.max(zeros, ones);
    }
}

File: Easy/Python/1700. Number of Students Unable to Eat Lunch.py
class Solution(object):
    def countStudents(self, students, sandwiches):
        zeros = 0
        ones = 0

        for n in students:
            if n == 0:
                zeros += 1
            else:
                ones += 1
        
        for n in sandwiches:
            if (n == 0 and zeros == 0) or (n == 1 and ones == 0):
                break
            if n == 0:
                zeros -= 1
            else:
                ones -= 1
        
        return max(zeros, ones)
